internal/jobservice/repository/sqlite.go

```diff
@@ -374,3 +367,35 @@ func (s *JSRepoSQLite) GetSubscribedJobSets(ctx context.Context) ([]SubscribedTu
 	}
 	return tuples, nil
 }
+
+func (s *JSRepoSQLite) purgeExpiredJobsAndJobSets() {
+	ticker := time.NewTicker(time.Duration(s.jobServiceConfig.PurgeJobSetTime) * time.Second)
+	log := log.WithField("JobService", "ExpiredJobSetsPurge")
+	jobsetsStmt := fmt.Sprintf(`DELETE FROM jobsets WHERE Timestamp < (UNIXEPOCH() - %d);`, s.jobServiceConfig.PurgeJobSetTime)
+	jobServiceStmt := fmt.Sprintf(`DELETE FROM jobservice WHERE Timestamp < (UNIXEPOCH() - %d);`, s.jobServiceConfig.PurgeJobSetTime)
+	tableExistsStmt := `SELECT EXISTS (SELECT 1 FROM sqlite_master WHERE type='table' AND name=?);`
+	for range ticker.C {
+		s.lock.Lock()
+		for _, table := range []string{"jobsets", "jobservice"} {
+			var exists bool
+			err := s.db.QueryRow(tableExistsStmt, table).Scan(&exists)
```

Comment:
This is a very robust check, but I think it may not really be necessary - if either of these tables doesn't exist, the DELETE operation will return an error that will loudly say if the table is missing. (And if that's the case, then the user of this code has bigger problems.) I'd just process with running the DELETE directly.

Reply:
I added exist-check because I saw some random error logs during testing. But the CI now is not affected by so, it will remove it.